operator/apis/platform/v1alpha1/central_types.go

```diff
@@ -166,6 +147,48 @@ func (c *CentralComponentSpec) GetAdminPasswordGenerationDisabled() bool {
 	return pointer.BoolPtrDerefOr(c.AdminPasswordGenerationDisabled, false)
 }
 
+// CentralDBSpec defines settings for the "central db" component.
+type CentralDBSpec struct {
+	// Specify a secret that contains the password in the "password" data item.
+	// If omitted, the operator will auto-generate a DB password and store it in the "password" item
+	// in the "central-db-password" secret.
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Administrator Password",order=1
+	PasswordSecret *LocalSecretReference `json:"passwordSecret,omitempty"`
+
+	// Disable admin password generation. Do not use this for first-time installations,
+	// as you will have no way to perform initial setup and configuration of alternative authentication methods.
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:hidden"}
+	PasswordGenerationDisabled *bool `json:"passwordGenerationDisabled,omitempty"`
+
+	// Specify a connection string that corresponds to an existing database. If set, the operator will not manage Central DB
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,order=2
+	ConnectionStringOverride *string `json:"connectionString,omitempty"`
```

Comment:
Thought about it this weekend and I think having this signify external database is a mistake because there may some circumstance when we want to tweak the connection string, but if this means external then there is not alternative